students/briggsm/session04/mb_trigram.py

```diff
@@ -1,54 +1,81 @@
+''''
+Matt Briggs Trigram Exercise
+Version 2.0 2017.11.19
+'''
+
 import random as rand
 
+
 def create_corpus(infile):
-    '''Input: a filename
-       Output: a list of words.'''
+    ''' Input: a filename of a text file
+        Output: a list of words from the text file.'''
     corpus = []
     with open(infile) as lines:
         lines.readline()
         for line in lines:
             line = line.strip()
             words = line.split(" ")
-            for i in range(len(words)):
-                corpus.append(words[i])
+            for word in enumerate(words):
+                corpus.append(word[1])
     return corpus
 
+
 def create_trigram(textlist):
-    '''Input: a list of words from a text body.
-       Output: a dictionary of two words in sequence and the word after.'''
+    ''' Input: a list of words from a text body.
+        Output: a dictionary of two words in sequence and the word after.'''
     trigram = {}
-    for w in range(len(textlist[:-2])):
-        firstword = textlist[w]
-        secondword = textlist[w+1]
-        thirdword = textlist[w+2]
-        trigram[thirdword] = (firstword, secondword)
+    for x in range(len(textlist[:-2])):
+        firstword = textlist[x]
+        secondword = textlist[x+1]
+        thirdword = textlist[x+2]
+        #print("[({},{}] = {}".format(firstword, secondword, thirdword))
+        if (firstword, secondword) in trigram:
```

Comment:
exactly right.

check out dict.setdefault() for a nifty way to reduce this to one line --but this is fine :-)